files/synchronize.py

```diff
@@ -345,15 +352,22 @@ def main():
     ssh_opts = '-S none'
 
     if not verify_host:
-      ssh_opts = '%s -o StrictHostKeyChecking=no' % ssh_opts
+        ssh_opts = '%s -o StrictHostKeyChecking=no' % ssh_opts
 
     if ssh_args:
-      ssh_opts = '%s %s' % (ssh_opts, ssh_args)
+        ssh_opts = '%s %s' % (ssh_opts, ssh_args)
 
     if dest_port != 22:
-        cmd += " --rsh 'ssh %s %s -o Port=%s'" % (private_key, ssh_opts, dest_port)
-    else:
-        cmd += " --rsh 'ssh %s %s'" % (private_key, ssh_opts)  # need ssh param
+        ssh_opts = '%s -o Port=%d' % (ssh_opts, dest_port)
+
+    if private_key:
+        cmd += " --rsh 'ssh %s %s'" % (private_key, ssh_opts)
+    elif ssh_passwd:
+        import os
+        sshpass_pipe = os.pipe()
+        os.write(sshpass_pipe[1], "{0}\n".format(ssh_passwd))
+        map(os.close, sshpass_pipe)
+        cmd += " --rsh 'sshpass -d %d ssh -l ass %s'" % (sshpass_pipe[0], ssh_opts)
```

Comment:
Does this work reliably ?  run_command ( b/lib/ansible/module_utils/basic.py +1591) creates a pipe and writes to it, and this (stdin fd ) number can be pretty random.

            st_in = subprocess.PIPE